circuits/scripts/withdraw.js

```diff
@@ -0,0 +1,130 @@
+    var ret = 0n;
+    var exp = 0n;
+    while (x > 0) {
+        var mod = x % (2n ** small_stride);
+        ret = ret + mod * (2n ** (stride * exp));
+        x = x / (2n ** small_stride);
+        exp = exp + 1n;
+    }
+    return ret;
+}
+
+// bigendian
+function bigint_to_Uint8Array(x) {
+    var ret = new Uint8Array(32);
+    for (var idx = 31; idx >= 0; idx--) {
+        ret[idx] = Number(x % 256n);
+        x = x / 256n;
+    }
+    return ret;
+}
+
+// bigendian
+function Uint8Array_to_bigint(x) {
+    var ret = 0n;
+    for (var idx = 0; idx < x.length; idx++) {
+        ret = ret * 256n;
+        ret = ret + BigInt(x[idx]);
+    }
+    return ret;
+}
+
+var privkeys = [88549154299169935420064281163296845505587953610183896504176354567359434168161n,
+                               37706893564732085918706190942542566344879680306879183356840008504374628845468n,
+                               90388020393783788847120091912026443124559466591761394939671630294477859800601n,
+                               110977009687373213104962226057480551605828725303063265716157300460694423838923n];
+var ethAddrsHex = ['0x4eaB0fA55a6c71c0c23BD0E67C06682b4Ce78A7f', '0xda1EA97d4c6D723125B26E4a1441C399d0EA0bE4', '0x05b9Df7825DE3c65A7D0386bD470F32a7008bC0c', '0xb032BA0CbF94CfE9A377aD4060624cA31CEEdE43'];
+var ethAddrsBigInt = [449116070504281332671503011463517494968310008447n, 1245243775008760871701837986296674110984394771428n, 32690058639829526695160611728147099830035463180n, 1005913620148149221978770586875249574770925035075n];
+
+stateTree.update(0, vmtree.poseidon([ethAddrsBigInt[0], 0n, 10n]));
+stateTree.update(1, vmtree.poseidon([ethAddrsBigInt[1], 12n, 15n]));
+stateTree.update(2, 0);
+stateTree.update(3, 0);
+stateTree.update(4, 0);
+
+console.log(stateTree.root);
+
+async function generateInput(privKey, sender, receiver, leafCoins, leafIndex) {
+    let msghash_bigint = vmtree.poseidon([leafCoins[0], leafCoins[1], receiver]);
+
+    // in compact format: r (big-endian), 32-bytes + s (big-endian), 32-bytes
+    var sig/*: Uint8Array*/ = await signAsync(bigint_to_Uint8Array(msghash_bigint), bigint_to_Uint8Array(privKey));
+    var pubKey = ProjectivePoint.fromPrivateKey(privKey);
+
+    const {pathIndices, pathElements} = stateTree.path(leafIndex);
+    const input = vmtree.utils.stringifyBigInts({
+        step_in: [stateTree.root],
+        sender: sender,
+        recipient: 0n,
+        leaf_coins: leafCoins,
+        pathElements: pathElements,
+        pathIndices: pathIndices,
+        r: bigint_to_array(64, 4, sig.r),
+        s: bigint_to_array(64, 4, sig.s),
+        msghash: bigint_to_array(64, 4, msghash_bigint),
+        pubkey: [bigint_to_array(64, 4, pubKey.x), bigint_to_array(64, 4, pubKey.y)]
+    });
+    return input;
+}
+
+// user0 sends [0,9] from first leaf to user1
+const input0 = await generateInput(privkeys[0], ethAddrsBigInt[0], 0n, [0n, 10n], 0);
```

Comment:
Doesn't it send `[0, 10]` inclusive?
Whenever I use the symbols `[x, y]`, I mean to include `x` and `y`, as per [wikipedia](https://en.wikipedia.org/wiki/Interval_(mathematics)#Including_or_excluding_endpoints).

<img width="309" alt="Screen Shot 2023-06-20 at 1 16 59 PM" src="https://github.com/BlakeMScurr/Altitude/assets/18047446/a42e978e-7484-41ec-b4d1-88de53eb7e0a">